core-bundle/src/Resources/contao/dca/tl_content.php

```diff
@@ -1079,6 +1085,89 @@ public function preserveReferenced()
 		}
 	}
 
+	/**
+	 * Filter the content elements
+	 */
+	public function filterContentElements()
+	{
+		if ($this->User->isAdmin)
+		{
+			return;
+		}
+
+		if (empty($this->User->elements))
+		{
+			$GLOBALS['TL_DCA']['tl_content']['config']['closed'] = true;
+			$GLOBALS['TL_DCA']['tl_content']['config']['notEditable'] = true;
+		}
+		else
+		{
+			// Set the first allowed content element as default value
+			$GLOBALS['TL_DCA']['tl_content']['fields']['type']['default'] = $this->User->elements[0];
+		}
```

Comment:
I don't think this change is correct. It would basically override the default value for all users. All users that are not admins will now get the `headline` element as default. In other words, the database has a default of `text` but we are always overriding it (DCA overwrites SQL). I see 2 possible solutions:
 1. use `Widget::getEmptyValueByFieldType()` or similar
 2. move the default for the content element type back to the DCA instead of the SQL (does not cause any harm)

Reply:
Fixed in 5e98690dbfdd866dbded99a7aed4e7bda73e04f5.